Cargo.toml

```diff
@@ -163,14 +163,14 @@ xshell = "0.2.5"
 dashmap = { version = "=5.5.3", features = ["raw-api"] }
 
 [workspace.lints.rust]
-bare_trait_objects = "warn"
+# remember to update RUSTFLAGS in ci.yml if you add something here
```

Comment:
Wonder if we can make this work out better for CI 🤔

Reply:
We could just set them to `deny`, I don't think it's going to be _too_ annoying.

I don't think we can customize them for specific profiles: https://users.rust-lang.org/t/changing-lint-configuration-based-on-cargo-profile/107286/2.

Reply:
> I don't think it's going to be too annoying.

That will stop check early won't it? That will absolutely annoy me when refactoring because I don't fix all of those warnings immediately

Reply:
That depends on whether you're using `--keep-going` or not :smile:.

Reply:
Well no, because any dependency that errors (due to this) will prevent dependents from building still 😅 At least that is my understanding of `--keep-going`, it will make cargo no bail out on other independent work.